pkg/util/overridemanager/overridemanager_test.go

```diff
@@ -14,6 +19,195 @@ import (
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "test clusterOverridePolicy",
+			fields: fields{
+				Client: fake.NewClientBuilder().WithScheme(gclient.NewSchema()).WithObjects(&clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "test2"},
+					Spec: clusterv1alpha1.ClusterSpec{
+						APIEndpoint:                 "https://127.0.0.1",
+						SecretRef:                   &clusterv1alpha1.LocalSecretReference{Namespace: "ns1", Name: "secret1"},
+						InsecureSkipTLSVerification: true,
+						ProxyURL:                    "http://1.1.1.1",
+					},
+				},
+					&corev1.Secret{
+						ObjectMeta: metav1.ObjectMeta{Namespace: "ns2", Name: "secret2"},
+						Data:       map[string][]byte{clusterv1alpha1.SecretTokenKey: []byte("token")},
+					}, clusterRoleObj, &policyv1alpha1.ClusterOverridePolicy{
+						ObjectMeta: metav1.ObjectMeta{Name: "test2", Namespace: metav1.NamespaceDefault},
+						Spec: policyv1alpha1.OverrideSpec{
+							ResourceSelectors: []policyv1alpha1.ResourceSelector{
+								{
+									APIVersion: "rbac.authorization.k8s.io/v1",
+									Kind:       "ClusterRole",
+									Name:       "test2",
+								},
+							},
+							OverrideRules: []policyv1alpha1.RuleWithCluster{
+								{
+									TargetCluster: &policyv1alpha1.ClusterAffinity{ClusterNames: []string{"test2"}},
+									Overriders: policyv1alpha1.Overriders{
+										LabelsOverrider: []policyv1alpha1.LabelAnnotationOverrider{
+											{
+												Operator: policyv1alpha1.OverriderOpAdd,
+												Value: map[string]string{
+													"test2": "test2",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				).Build(),
+				EventRecorder: &record.FakeRecorder{},
+			},
+			args: args{
+				rawObj:      clusterRoleObj,
+				clusterName: "test2",
+			},
+			want: &AppliedOverrides{
+				AppliedItems: []OverridePolicyShadow{
+					{
+						PolicyName: "test2",
+						Overriders: policyv1alpha1.Overriders{
+							LabelsOverrider: []policyv1alpha1.LabelAnnotationOverrider{
+								{
+									Operator: policyv1alpha1.OverriderOpAdd,
+									Value: map[string]string{
+										"test2": "test2",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want1:   nil,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			o := &overrideManagerImpl{
+				Client:        tt.fields.Client,
+				EventRecorder: tt.fields.EventRecorder,
+			}
+			got, got1, err := o.ApplyOverridePolicies(tt.args.rawObj, tt.args.clusterName)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ApplyOverridePolicies() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ApplyOverridePolicies() got = %v, want %v", got, tt.want)
+			}
+			if !reflect.DeepEqual(got1, tt.want1) {
+				t.Errorf("ApplyOverridePolicies() got1 = %v, want %v", got1, tt.want1)
+			}
```

Comment:
After applying override policies, the original resource has changed meanwhile, maybe we can check it at the same time.

Reply:
ok